src/main/java/org/dbunit/dataset/builder/DataSetBuilder.java

```diff
@@ -45,11 +46,17 @@ public DataSetBuilder() throws DataSetException {
     }
 
     public DataSetBuilder(boolean ignoreCase) throws DataSetException {
-        this(stringPolicy(ignoreCase));
-    }
-
-    public DataSetBuilder(IStringPolicy stringPolicy) throws DataSetException {
-        this.stringPolicy = stringPolicy;
+    	//TODO create an issue in DBUnit so that CachedDataSet has a constructor that receives _caseSensitiveTableNames (without IDataSetConsumer).
+    	Field caseSensitiveField;
+		try {
+			caseSensitiveField = AbstractDataSet.class.getDeclaredField("_caseSensitiveTableNames");
+			caseSensitiveField.setAccessible(true);
+	    	caseSensitiveField.set(this.dataSet, !ignoreCase);
+		} catch (Exception e) {
+			//ignore. It shouldn't happen at runtime, unless there's programming error.
```

Comment:
If it could never happend you should throw a RuntimeException with a clear message text.